### PR TITLE
Switch to hugo modules

### PR DESCRIPTION
**Pull request by antonia-avramova**
Switch to hugo modules

Signed-off-by: Antonia Avramova <antonia.avramova@bosch.io>

### CODE REVIEW

File: web/site/go.mod [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module github.com/eclipse-kanto/kanto
+
+go 1.22.0
```
Comment (dimitar-dimitrow) [inline web/site/go.mod:3]: Could we use go 1.21 be used as a minimal version? If there is no explicit dependency for 1.22, we better keep 
 to the older version, as 1.21 is the most used version for builds across kanto repositories.

File: web/site/content/docs/references/containers/api-reference/_index.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+title: "API Reference"
```
Comment (dimitar-dimitrow) [inline web/site/content/docs/references/containers/api-reference/_index.md:2]: This `API Reference` would fit better in a separate PR. @k-gostev what do you think?

**Review by dimitar-dimitrow — CHANGES_REQUESTED** [review]
This [readme](https://github.com/eclipse-kanto/kanto/blob/main/web/README.md) should also be updated according to other changes from this PR:
- the sub-module step seems unnecessary, as we switch to Hugo modules
- the minimal Hugo version should be updated
- the link for installing Hugo may be updated
- kanto logo link is wrong

File: web/site/content/docs/references/containers/api-reference/_index.md (hunk 2 above)
Comment (k-gostev) [reply web/site/content/docs/references/containers/api-reference/_index.md:2]: Absolutely, this commit was due to some tests that I made, so it should be dropped

**Comment by dimitar-dimitrow** [comment]
@k-gostev, @antonia-avramova  Do we need to update the Hugo version in the [documentation](https://github.com/eclipse-kanto/kanto/blob/main/.github/workflows/documentation.yaml) workflow as well?

**Review by dimitar-dimitrow — APPROVED** [review]

**Pull request merged**